pkg/endpoint/endpoint.go

```diff
@@ -428,15 +428,18 @@ type Endpoint struct {
 	// LabelsMap is the Set of all security labels used in the last policy computation
 	LabelsMap *identityPkg.IdentityCache
 
-	// Consumable represents the security-identity-based policy for this endpoint.
-	Consumable *policy.Consumable `json:"-"`
+	// Iteration policy of the Endpoint - what does that even mean??
+	Iteration uint64 `json:"-"`
 
-	// L4Policy is the L4Policy in effect for the
+	// RealizedL4Policy is the L4Policy in effect for the
 	// endpoint. Outside of policy recalculation, it is the same as the
-	// Consumable's L4Policy, but this is needed during policy recalculation to
+	// Consumable's RealizedL4Policy, but this is needed during policy recalculation to
 	// be able to clean up PolicyMap after the endpoint's consumable has already
 	// been updated.
-	L4Policy *policy.L4Policy `json:"-"`
+	RealizedL4Policy *policy.L4Policy `json:"-"`
+
+	// DesiredL4Policy is the desired L4Policy for the endpoint.
+	DesiredL4Policy *policy.L4Policy `json:"-"`
```

Comment:
In my next (future) upload I've specified that it is populated during policy generation - I don't want to get too specific specifying where things are called because things like that get stale.

Reply:
They do get stale, but that is actually a failure mode. One can just grep and find where this gets set. What isn't apparent is the intent, and if the comment is stale that means we made a change and didn't fix the comment. This is bad apart from the stale comment (the code is now also potentially stale, it just happens to work).
Still, I also dislike stale comments so the question really is whether our code-quality is good enough to make such a comment useful or a liability. You're suggesting it isn't, and I, sadly, agree.